### PR TITLE
Added abundance_colocalization_plot function

### PR DESCRIPTION
**Pull request by ptajvar**
## Description

Adding a plot function: abundance_colocalization_plot . This function accepts a PXL object and two sets of markers, markers_x and markers_y, and makes a scatter plot of marker abundances of all pair combinations of the two sets. Each point in the scatter plot corresponds to one component in the PXL object and is colored based on the colocalization of the two markers in the corresponding plot and sized based on the absolute value of this colocalization. 

Fixes: EXE-1852

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

A new unit-test is added for making an example plot.

## PR checklist:

- [x] This comment contains a description of changes (with reason).
- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked my code and documentation and corrected any misspellings
- [x] I have documented any significant changes to the code in [CHANGELOG.md](../CHANGELOG.md)

### CODE REVIEW

File: src/pixelator/plot/__init__.py [hunk 1]
```diff
@@ -454,3 +454,74 @@ def density_scatter_plot(
...
+    # TODO: See how the legend is determined based on the merged data to be
+    # able to access actual marker sizes.
+    for i in range(1, 6):
+        if i >= len(plot_grid._legend_data):
+            break
+        plot_grid._legend_data[list(plot_grid._legend_data.keys())[i]].set_markersize(5)
+    legend_data = {
+        i: plot_grid._legend_data[i] for i in list(plot_grid._legend_data.keys())[:6]
+    }
+    plot_grid.add_legend(legend_data=legend_data)
```
Comment (johandahlberg) [inline src/pixelator/plot/__init__.py:526]: I'm not sure I understand what this is trying to achieve, maybe we can put a small comment on what this is trying to achive?

**Review by johandahlberg — APPROVED** [review]
I had a request for clarification on one thing, otherwise this lgtm.

File: tests/snapshots/test_plot/test_abundance_colocalization_plot/test_abundance_colocalization_plot.png
Comment (ambarrio) [inline tests/snapshots/test_plot/test_abundance_colocalization_plot/test_abundance_colocalization_plot.png:1]: Why is this Figure empty as a plot?

**Review by ambarrio — COMMENTED** [review]
Just a small comment post-review. I saw a figure empty and I though something was wrong.

File: tests/snapshots/test_plot/test_abundance_colocalization_plot/test_abundance_colocalization_plot.png
Comment (ptajvar) [reply tests/snapshots/test_plot/test_abundance_colocalization_plot/test_abundance_colocalization_plot.png:1]: The pixel we get from setup_basic_pixel_dataset is very small and we're plotting incompatible markers as a way to look for doublets so there's only one component that has both CD3 and CD19 resulting in a single point in the lower left plot and other plots remaining empty.

Comment (ambarrio) [reply tests/snapshots/test_plot/test_abundance_colocalization_plot/test_abundance_colocalization_plot.png:1]: Aha! I understand this now.

**Pull request merged**